File: src/main/scala/org/galaxio/gatling/kafka/protocol/KafkaProtocolBuilderBackwardCompatible.scala
package org.galaxio.gatling.kafka.protocol

import org.apache.kafka.clients.producer.ProducerConfig
import org.galaxio.gatling.kafka.protocol.KafkaProtocol.KafkaKeyMatcher

import scala.concurrent.duration.FiniteDuration

final case class KafkaProtocolBuilderBackwardCompatible(topic: String, props: Map[String, AnyRef], timeout: FiniteDuration) {

  def build: KafkaProtocol = {
    val serializers = Map(
      ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG   -> "org.apache.kafka.common.serialization.ByteArraySerializer",
      ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG -> "org.apache.kafka.common.serialization.ByteArraySerializer",
    )
    KafkaProtocol(topic, props ++ serializers, Map.empty, timeout, KafkaKeyMatcher)
  }

}

File: src/main/java/org/galaxio/gatling/kafka/javaapi/protocol/KafkaProtocolBuilderBackwardCompatible.java
package org.galaxio.gatling.kafka.javaapi.protocol;

import io.gatling.core.protocol.Protocol;
import io.gatling.javaapi.core.ProtocolBuilder;

public class KafkaProtocolBuilderBackwardCompatible implements ProtocolBuilder {

    private final org.galaxio.gatling.kafka.protocol.KafkaProtocolBuilderBackwardCompatible wrapped;

    public KafkaProtocolBuilderBackwardCompatible(org.galaxio.gatling.kafka.protocol.KafkaProtocolBuilderBackwardCompatible wrapped) {
        this.wrapped = wrapped;
    }

    @Override
    public Protocol protocol() {
        return wrapped.build();
    }

}
